sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -62,31 +62,58 @@ export class ApplicationControllerService {
     data: ApplicationData,
   ): Promise<ApplicationFormData> {
     const additionalFormData = {} as ApplicationFormData;
-    // Check wether the selected location is designated or not.
-    // If selected location is not designated, then make the
-    // selectedLocation null
+    await this.processSelectedLocation(data, additionalFormData);
+    await this.processSelectedProgram(data, additionalFormData);
+    await this.processSelectedOffering(data, additionalFormData);
+    return { ...data, ...additionalFormData };
+  }
+
+  /**
+   * Check whether the selected location is designated or not.
+   * If selected location is not designated, then make the
+   * selectedLocation null.
+   * @param data application data.
+   * @param additionalFormData form data for readonly form.
+   */
+  private async processSelectedLocation(
+    data: ApplicationData,
+    additionalFormData: ApplicationFormData,
+  ) {
     if (data.selectedLocation) {
       const designatedLocation = await this.locationService.getLocation(
         data.selectedLocation,
       );
       if (!designatedLocation.isDesignated) {
         data.selectedLocation = null;
       }
-      // Assign location name for readonly form
+      // Assign location name for readonly form.
       additionalFormData.selectedLocationName = designatedLocation.locationName;
     }
-    // Check wether the program is approved or not.
-    // If selected program is not approved, then make the
-    // selectedLocation null
+  }
+
+  /**
+   * Check wether the program is approved or not.
+   * If selected program is not approved, then make the
+   * selectedProgram null.
+   * @param data application data.
+   * @param additionalFormData form data for readonly form.
+   */
+  private async processSelectedProgram(
+    data: ApplicationData,
+    additionalFormData: ApplicationFormData,
+  ) {
     if (data.selectedProgram) {
       const selectedProgram = await this.programService.getProgramById(
         data.selectedProgram,
       );
-
       if (selectedProgram) {
-        // Assign program name for readonly form
+        // Assign selected program && selected offering for application as null when the program is inactive.
+        if (!selectedProgram.isActive) {
+          data.selectedProgram = null;
+        }
```

Comment:
Minor. This condition can be moved after this as they both are relatively same context.

![image](https://github.com/bcgov/SIMS/assets/54600590/64cb0632-4a18-4a27-8d29-ec29ff6fb7e3)